modules/charon-core/src/main/java/org/wso2/charon3/core/config/SCIMSystemSchemaExtensionBuilder.java

```diff
@@ -156,4 +180,15 @@ protected boolean isRootConfig(ExtensionAttributeSchemaConfig config) {
 
         return StringUtils.equals(extensionRootAttributeName, config.getName());
     }
+
+    private void readConfiguration(List<SCIMCustomAttribute> schemaConfigurations, Map<String,
+            ExtensionAttributeSchemaConfig> attributeConfigs) {
+
+        for (SCIMCustomAttribute schemaConfiguration : schemaConfigurations) {
+            ExtensionAttributeSchemaConfig schemaAttributeConfig =
+                    new ExtensionAttributeSchemaConfig
+                            (schemaConfiguration.getProperties());
+            attributeConfigs.put(schemaAttributeConfig.getURI(), schemaAttributeConfig);
```

Comment:
if (schemaAttributeConfig.getURI().startsWith(EXTENSION_ROOT_ATTRIBUTE_URI)) {
     extensionConfig.put(schemaAttributeConfig.getURI(), schemaAttributeConfig);
}

Better to add this check since this readConfiguration method is called from public buildSystemSchemaExtension method, so we can not guarantee only system schema related attributes are sent there.